connectors/src/connectors/microsoft/temporal/workflows.ts

```diff
@@ -47,9 +50,21 @@ export async function fullSyncWorkflow({
 }) {
   await syncStarted(connectorId);
 
-  if (nodeIdsToSync === undefined) {
-    nodeIdsToSync = await getRootNodesToSync(connectorId);
-  }
+  setHandler(folderUpdatesSignal, (folderUpdates: FolderUpdatesSignal[]) => {
```

Comment:
not sure I get why we add signal handling to the workflow; can't see where signals are sent and why they are needed VS sending the permission diff to setPermissions when doing add/remove?
Will ask you IRL so I better understand

Reply:
We're using signalWithStart , which starts the workflow if it's not running, or send a signal if it does.